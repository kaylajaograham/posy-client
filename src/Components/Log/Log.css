
.Log {
    --note-height: 88px;
    background-color: darkseagreen;
    color: black;
    height: var(--note-height);
    margin: 50px 20px;
    padding: 8px 0 8px 8px;
    position: relative;
    border-radius: 25px;
  
  }
  
  .Log__delete {
    --note-button-height: 32px;
    background-color: peru;
    border: 0;
    border-bottom-left-radius: 25px;
    border-bottom-right-radius: 25px;
    color: white;
    float: right;
    font-size: 12px;
    height: var(--note-button-height);
    position: relative;
    top: calc((var(--note-button-height) + 32px) * -1);
    transform-origin: right bottom;
    transform: rotate(-90deg);
    width: calc(var(--note-height));
  }
  .Log__delete:hover {
    background-color: #bd0909;
    color: white;
    cursor: pointer;
    font-weight: bold;
  }
  
  .Log__name {
    font-family: var(--custom-font);
    background-color: darkseagreen;
    color: white;
    font-size: 20px;
    margin: 0;
    max-width: 90%;
    min-height: 24px;
    text-align: center;
  }
  .Log__name a {
    background-color: darkseagreen;
    color: white;
    display: block;
    text-decoration: none;
    
  }
  .Log__name a:hover {
    color: #FFB72F;
  }
  
  