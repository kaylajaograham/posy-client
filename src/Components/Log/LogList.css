
.LogList {
    background-color: #efefef;
    height: 100%;
    padding: 10px;
    width: 100%;
  }
  
  .LogList__content {
    color: goldenrod;
    max-width: 380px;
  }
  