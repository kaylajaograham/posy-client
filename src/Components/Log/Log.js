import React from 'react'
import { Link } from 'react-router-dom'

import './Log.css';
import PosyContext from '../../PosyContext';
import config from '../../config';



export default class Log extends React.Component {
  static defaultProps = {
    onDeleteLog: () => {},
  }

  static contextType = PosyContext;

  handleClickDelete = () => {
    const logId = this.props.id


    fetch(`${config.API_ENDPOINT}/logs/${logId}`, {
      method: 'DELETE',
      headers: {
        'content-Type': 'application/json'
      },
    })
    .then(res => {
      if (!res.ok) {
        return res.json().then(e => Promise.reject(e))
      }
    })
    .then(() => {
      // deletes from context which updates state by removing log
      this.context.deleteLog(logId)
      // sends user back to homepage after delete
      this.props.onDeleteLog(logId)
    })
    .catch(err => {
      console.log({ err })
    })
  }

  render() {
    const { log_name, id } = this.props

    return (
      <div className='Log'>
        <h2 className='Log__name'>
          <Link to={`/logs/${id}`}>
            {log_name}
          </Link>
        </h2>
        <button 
          className='Log__delete' 
          type='button'
          onClick={ this.handleClickDelete }>
          {' '}
          delete
        </button>
      </div>        
    )
  }
}
