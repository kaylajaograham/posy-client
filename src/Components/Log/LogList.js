import React from 'react'

import './LogList.css';

import PosyContext from '../PosyContext'
import Log from '.Log';
import { findLog } from '../../log-helpers';

export default class LogList extends React.Component {
  state = {
    forErrors: this.props.match,
    toggle: true
  }

  static defaultProps = {
    match: {
      params: {}
    }
  }

  static contextType = PosyContext;

  // sends user back to homepage after delete
  handleDeleteLog = logId => {
    this.props.history.push('/')
  }
    

  render () {
    const { logs=[] } = this.context
    const { logId } = this.state.forErrors.params
    const log = findLog(logs, logId) || { content: ''}
      if(this.state.toggle === false) {
        this.setState({
          forErrors: 'err'
        })
        this.setState({
          forErrors: this.props.match
        })
      }

    return (
      <section className='LogList'>
        <Log
          id={log.id}
          log_name={log.log_name}
          log_tag={log.log_tag}
          log_entry={log.log_entry}
          onDeleteLog={this.handleDeleteLog}
        />
        <div className='LogList__content'>
          {log.content.split(/\n \r|\n/).map((para, i) =>
            <p key={i}>{para}</p>
          )}
        </div>
      </section>
    )    
  }

}
