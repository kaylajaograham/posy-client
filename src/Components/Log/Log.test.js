import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Log from './Log';

describe(`AddActivityItemForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders a form.AddActivityItemForm by default', () => {
    const wrapper = shallow(<Log />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the AddActivityItemForm given props', () => {
    const wrapper = shallow(<Log {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})