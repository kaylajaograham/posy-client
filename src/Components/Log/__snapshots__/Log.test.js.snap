// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddActivityItemForm component renders a form.AddActivityItemForm by default 1`] = `
<div
  className="Log"
>
  <h2
    className="Log__name"
  >
    <Link
      to="/logs/undefined"
    />
  </h2>
  <button
    className="Log__delete"
    onClick={[Function]}
    type="button"
  >
     
    delete
  </button>
</div>
`;

exports[`AddActivityItemForm component renders the AddActivityItemForm given props 1`] = `
<div
  className="Log"
>
  <h2
    className="Log__name"
  >
    <Link
      to="/logs/undefined"
    />
  </h2>
  <button
    className="Log__delete"
    onClick={[Function]}
    type="button"
  >
     
    delete
  </button>
</div>
`;
